kernel/src/main/java/org/kframework/backend/kore/KoreBackend.java

```diff
@@ -227,43 +230,97 @@ public Function<Definition, Definition> steps() {
+            entry("resolveFreshConstants", resolveFreshConstants),
+            entry("generateSortPredicateSyntax2", generateSortPredicateSyntax),
+            entry("generateSortProjections2", generateSortProjections),
+            entry("checkSimplificationRules", checkSimplificationRules),
+            entry("subsortKItem2", subsortKItem),
+            entry(
+                "addStrategyCellToRules",
+                d -> new Strategy().addStrategyCellToRulesTransformer(d).apply(d)),
+            entry(
+                "addStrategyRuleToMainModule",
+                d -> Strategy.addStrategyRuleToMainModule(d.mainModule().name()).apply(d)),
+            entry("concretizeCells", d -> ConcretizeCells.transformDefinition(d)),
+            entry("genCoverage", genCoverage),
+            entry("addSemanticsModule", Kompile::addSemanticsModule),
+            entry("resolveConfigVar", resolveConfigVar),
+            entry("addCoolLikeAtt", addCoolLikeAtt),
+            entry("markExtraConcreteRules", markExtraConcreteRules),
+            entry("removeAnywhereRules", removeAnywhereRules),
+            entry("generateSortPredicateRules", generateSortPredicateRules),
+            entry("numberSentences2", numberSentences));
+
+    List<String> stepOrdering;
+    if (kompileOptions.koreBackendSteps != null) {
+      stepOrdering = List.of(kompileOptions.koreBackendSteps.split(","));
+    } else {
+      stepOrdering =
+          List.of(
+              "resolveComm",
+              "resolveIO",
+              "resolveFun",
+              "resolveFunctionWithConfig",
+              "resolveStrict",
+              "resolveAnonVars",
+              "resolveContexts",
+              "numberSentences1",
+              "resolveHeatCoolAttribute",
+              "resolveSemanticCasts",
+              "subsortKItem1",
+              "constantFolding",
+              "propagateMacroToRules",
+              "guardOrs",
+              "resolveFreshConfigConstants",
+              "generateSortPredicateSyntax1",
+              "generateSortProjections1",
+              "expandMacros",
+              "addImplicitComputationCell",
+              "resolveFreshConstants",
+              "generateSortPredicateSyntax2",
+              "generateSortProjections2",
+              "checkSimplificationRules",
+              "subsortKItem2",
+              "addStrategyCellToRules",
+              "addStrategyRuleToMainModule",
+              "concretizeCells",
+              "genCoverage",
+              "addSemanticsModule",
+              "resolveConfigVar",
+              "addCoolLikeAtt",
+              "markExtraConcreteRules",
+              "removeAnywhereRules",
+              "generateSortPredicateRules",
+              "numberSentences2");
+    }
+
+    Function1<Definition, Definition> applyPipeline =
+        stepOrdering.stream()
+            .map(name -> namedStages.get(name))
```

Comment:
Let's add a short error message if the name doesn't exist - current behavior will just give a null pointer exception.